File: src/omnia/data/structure/rx/WritableObservableList.java
package omnia.data.structure.rx;

import omnia.data.contract.Writable;
import omnia.data.structure.mutable.MutableList;

public interface WritableObservableList<E>
    extends MutableList<E>, ObservableList<E>, Writable<ObservableList<E>> {

  static <E> WritableObservableList<E> create() {
    return new WritableObservableListImpl<>();
  }
}

File: omnia/src/commonMain/omnia/data/structure/observable/writable/WritableObservableList.kt
package omnia.data.structure.observable.writable

import omnia.data.contract.Writable
import omnia.data.structure.mutable.MutableList
import omnia.data.structure.observable.ObservableList

interface WritableObservableList<E : Any> :
    MutableList<E>, ObservableList<E>, Writable<ObservableList<E>> {

  companion object {

    fun <E : Any> create(): WritableObservableList<E> {
      return WritableObservableListImpl()
    }
  }
}